### DISCUSSION TITLE
Tenant Migrations in Background

### DISCUSSION THREAD

**Discussion by gitkwarteng** [django-tenants/django-tenants#977] [Q&A] [answered]
I am running django, uwsgi and nginx. my application is growing big with a lot of migration files. When a new tenant is created, migrations for new tenant takes a very long time, and it blocks server. I tried using usgi_tasks to handle tenant.create_schema() in the background, but migrations don't run. is there a way to go about it?

**Comment by tomturner** [comment]
Try cloning a tenant instead of creating a new one

**Reply by gitkwarteng** [reply]
thanks for the reply, can you please direct me on how to go about that? thanks.

**Reply by gitkwarteng** [reply]
ok, so I figured clone_tenant is one of the management commands in django_tenants. will try that and give feedback. thanks.

**Reply by gitkwarteng** [reply]
it appears clone_schema clones data in schema as well, is there a way to prevent this?

**Comment by tomturner** [comment, marked as answer]
Not without modifying the code.  You could create a blank tenant to clone.

**Reply by gitkwarteng** [reply]
ok ... thanks. 

created a blank schema, but am getting this error:
Action: Tables  Diagnostics: line=SQL statement "ALTER INDEX gladys.product_productuom_product_id_idx1 RENAME TO product_pro_product_311ca2_idx;"
PL/pgSQL function clone_schema(text,text,boolean,boolean) line 290 at EXECUTE. 42P07. relation "product_pro_product_311ca2_idx" already exists
CONTEXT:  PL/pgSQL function clone_schema(text,text,boolean,boolean) line 685 at RAISE

**Reply by gitkwarteng** [reply]
so my issue is resolved. blank schema seems to be working. the issue was as a result of an index I had created through the model's meta. removing the index solved the error. thanks.

**Reply by tomturner** [reply]
The other way creating new tenants is to use celery.  This the way I do it.

**Reply by gitkwarteng** [reply]
ok ... am using uwsgi_task instead of celery. But it appears migrations don't run when task executes in background. the cloning is much faster compared to running migrations, so I will stick to that for now.
